lib/XMLHttpRequest.js

```diff
@@ -391,7 +452,61 @@ function XMLHttpRequest(opts) {
     }
 
     // Load files off the local filesystem (file://)
+    // or data from Data URI (data:)
     if (local) {
+      if (datauri) try {
+        // Triming from original url string since 
+        var data = url.href.substr(5, settings.url.length);
```

Comment:
`substr` is a deprecated legacy Internet Explorer String method, we should probably use either `substring` or `slice` here. Do note that both use an end index instead of the length as a second parameter.

Reply:
I use `substr` as i could see the functionality being used in other parts of the function. However again, i could change to `slice` or `substring`, need to check with the method supports wider range of node versions.

Reply:
I changed every usage of `substr` to `slice`